src/components/ebay-combobox/index.js

```diff
@@ -0,0 +1,179 @@
+        if (this.activeDescendant) {
+            this.activeDescendant.destroy();
+        }
+
+        if (this.expander) {
+            this.expander.cancelAsync();
+        }
+    },
+    handleExpand() {
+        elementScroll.scroll(this.getEls('option')[this.state.selectedIndex]);
+        this.moveCursorToEnd();
+        emitAndFire(this, 'combobox-expand');
+    },
+    handleCollapse() {
+        emitAndFire(this, 'combobox-collapse');
+    },
+    moveCursorToEnd() {
+        const currentInput = this.getEl('input');
+
+        if (currentInput) {
+            const len = currentInput.value.length;
+            currentInput.setSelectionRange(len, len);
+        }
+    },
+    handleComboboxKeyUp(originalEvent) {
+        const selectedEl = this.getEl('options').querySelector('.combobox__option--active');
+        let newValue = this.getEl('input').value;
+
+        eventUtils.handleUpDownArrowsKeydown(originalEvent, () => {
+            if (!this.expander.isExpanded() && this.getEls('option').length > 0) {
+                this.expander.expand();
+            }
+            this.moveCursorToEnd();
+        });
+
+        eventUtils.handleEnterKeydown(originalEvent, () => {
+            newValue = selectedEl && selectedEl.textContent || newValue;
+            this.setState('currentValue', newValue);
+            this.setSelectedIndex();
+            if (selectedEl) {
+                this.emitChangeEvent();
+            }
+        });
+
+        eventUtils.handleEscapeKeydown(originalEvent, () => {
+            this.expander.collapse();
+        });
+
+        eventUtils.handleTextInput(originalEvent, () => {
+            this.setState('currentValue', newValue);
+            this.setSelectedIndex();
+            this.emitChangeEvent();
+            this.toggleListbox();
+        });
+    },
+    handleComboboxBlur(evt) {
+        const wasClickedOption = this.getEls('option').some(option => option === evt.relatedTarget);
+
+        if (wasClickedOption) {
+            this.getEl('input').focus();
+        }
+
+        if (this.expander.isExpanded() && !wasClickedOption) {
+            this.emitChangeEvent();
+            this.expander.collapse();
+        }
+    },
+    handleOptionClick(evt) {
+        const selectedEl = evt.target.nodeName === 'DIV' ? evt.target : evt.target.parentNode;
+
+        this.setState('currentValue', selectedEl.textContent);
+        this.setSelectedIndex();
+        this.emitChangeEvent();
+        this.expander.collapse();
+    },
+    setSelectedIndex(index = 0) {
+        const newIndex = index || findIndex(this.state.options, option => option.text === this.state.currentValue);
+
+        this.setState('selectedIndex', newIndex);
+    },
+    emitChangeEvent() {
+        const selectedValue =
+            this.state.options[this.state.selectedIndex]
+            && this.state.options[this.state.selectedIndex].value
+            || '';
+
+        emitAndFire(this, 'combobox-change', {
```

Comment:
To avoid possible confusion with a text input's `onchange` event (i.e. it fires when input loses focus), how about we name this `combobox-autocomplete` instead?

Reply:
Just to clarify, we should only fire a `combobox-change` event when the combobox loses focus. If you are already doing that, then please ignore my comment.